src/Polly.Core/CircuitBreaker/Controller/CircuitStateController.cs

```diff
@@ -307,8 +307,8 @@ private void SetLastHandledOutcome_NeedsLock(Outcome<T> outcome)
 
     private BrokenCircuitException CreateBrokenCircuitException() => _breakingException switch
     {
-        Exception exception => new BrokenCircuitException(BrokenCircuitException.DefaultMessage, exception),
-        _ => new BrokenCircuitException(BrokenCircuitException.DefaultMessage)
+        Exception exception => new BrokenCircuitException(BrokenCircuitException.DefaultMessage, exception, _blockedUntil - _timeProvider.GetUtcNow()),
+        _ => new BrokenCircuitException(BrokenCircuitException.DefaultMessage, _blockedUntil - _timeProvider.GetUtcNow())
     };
```

Comment:
Rather than do the same thing in both branches, let's compute it once:

```
{
    var retryAfter = _blockedUntil - _timeProvider.GetUtcNow();
    return _breakingException switch
    {
        Exception exception => new BrokenCircuitException(BrokenCircuitException.DefaultMessage, exception, retryAfter),
        _ => new BrokenCircuitException(BrokenCircuitException.DefaultMessage, retryAfter),
    };
}
```